### DISCUSSION TITLE
Question about fitting SED of obscured AGN

### DISCUSSION THREAD

**Discussion by TianmuGao** [mboquien/cigale#4] [Q&A] [answered]
Hi!

I'm now using X-CIGALE to fit some SEDs, and I have some questions.

What I'm doing is to use a small aperture to measure the SED of some nearby obscured/unobscured SMBHs, which means the emission of the host galaxy is almost not included. For each object, I have X-ray, optical, NIR and sub-millimeter data-points. Temporarily I'm using sfhdelayed, bc03, nebular, dustatt_modified_starburst, casey2012, skirtor2016, xray and redshifting, and I borrow the parameter configuration from others' paper.

<img width="332" alt="Aperture" src="https://user-images.githubusercontent.com/58971624/141314981-65dde403-7f4e-48d3-aad0-2fe922a54d7b.png">

The question is that I find it hard to fit the SEDs of obscured SMBHs using 'normal' parameter configuration. Those sub-millimeter data-points are especially not well fitted. Thus, I want some instructions about how to set those parameters. I know the nuclear region can be really dusty, so I'm trying to enhance the attenuation at short wavelength and the dust emission, but it doesn't work.

![SED](https://user-images.githubusercontent.com/58971624/141315281-57ef7aea-5bb5-445b-b9b0-aeb5a8a9f45c.jpg)

Besides, I find that CIGALE calculates the emission of the host galaxy first, then adds the emission of the AGN. So the emission of the AGN is seemingly not reprocessed by the input dust component. But I'm measuring the integrated emission of the nuclear region of the galaxy. In this situation, obviously the emission of the AGN is absorbed, then re-emitted by not only the intrinsic polar dust, but also the extrinsic dust. So I'm not sure if X-CIGALE can be used to fit the SED of obscured AGN.

Thanks in advance.

**Comment by Guang91** [comment, marked as answer]
Could you upload your pcigale.ini file? 

AGN has its own dust component (torus + polar dust) in CIGALE. The torus is close to AGN itself, but the polar dust can be close to or far from the AGN. We keep AGN polar-dust independent from host-galaxy dust, because there's no reason to assume they're the same.  
Why do you think it doesn't work for your case?

**Reply by TianmuGao** [reply]
[pcigale.ini.txt](https://github.com/mboquien/cigale/files/7521934/pcigale.ini.txt)

Here is the .ini file corresponding to the figure above. This is not the latest version. I also tried to change the parameter of the polar dust.

[pcigale.ini.txt](https://github.com/mboquien/cigale/files/7521981/pcigale.ini.txt)

Here is the latest version.

![SED](https://user-images.githubusercontent.com/58971624/141337798-46f6fb33-36a7-46a2-b0d4-42cd644fe727.jpg)

I don't think it changes a lot. I mean the main contributor of the obscuration in this case should be the host-galaxy dust, not the polar dust.

**Reply by Guang91** [reply]
A few suggestions:
1. Your parameters of skirtor2016 look quite simple. Why do you fix i=30? Is that because your source is type 1? Also, why do you leave polar-dust EBV=0.1? Since you think the AGN obscuration is similar to the galaxy, you should set AGN EBV similar to the galaxy EBV. 

2. I'm not sure "casey2012" is appropriate here, as it's a rather simple model with black-body like spectrum. I suggest trying some other models like dale2014, dl2014, or themis. 

3. For xray, why do you set gam=1.0? Note that gam here means *intrinsic* photon index, which is typically ~1.8. gam=1.0 is very unusual. Also, have you corrected absorption for your input X-ray fluxes, as X-CIGALE needs intrinsic X-ray fluxes as input?  

4. In a few places, you can simplify your models to save time. You have a Z=0.05 metallicity. Is that really necessary? You have several  E_BV_factor. Since line emission is not important to your broad-band photometry, I'd just use one (maybe E_BV_factor=1 for simplicity). Also, I'm not sure if it's necessary to have multiple "powerlaw_slope".

**Reply by TianmuGao** [reply]
As you can see, I'm having few data-points, so I'm not sure if I can constrain the viewing angle. Actually, I have no clue to the viewing angle. And I have changed the AGN EBV in the second .ini file.

As for the X-ray photon index, it came from hard x-ray spectrum fitting. I have corrected the Galactic absorption. Also, the quality of the spectrum is insufficient for detailed fitting.

**Reply by Guang91** [reply]
For viewing angle, you can use 30 (type 1) and 70 (type 2). 
Is gam=1.0 the intrinsic one (after absorption correction)? That would be unusual. 
I guess it's the *apparent* photon index.

**Reply by TianmuGao** [reply]
Hi! Here is my latest attempt. I think it's much better than before. What I have been doing is to decrease AGN fraction, and now it has reached 0.01. So maybe this galaxy doesn't host an AGN. I think I still need to change the parameter configuration to improve the result of the fitting. Then I will try to find a parameter configuration, which can be used to fit other SEDs.

[pcigale.ini.txt](https://github.com/mboquien/cigale/files/7539368/pcigale.ini.txt)

![SED_1115](https://user-images.githubusercontent.com/58971624/141800971-1122eab2-2d86-4b98-8176-67bdfe61af07.jpg)

Meanwhile, I still have some questions. Since there is no ALMA filter in X-CIGALE filter repository,  I simply use the format of the VLA filters in X_CIGALE filter repository and change the wavelength coverage. Is it correct? I know it's only an approximation.

[ALMA.Band7.dat.txt](https://github.com/mboquien/cigale/files/7539497/ALMA.Band7.dat.txt)

I also would like to know how I can use X-CIGALE to plot the theoretical SED based on a specific set of parameters directly. I mean this may help me know how to change the parameter configuration.

Thank you!

**Reply by Guang91** [reply]
Thanks. Yeah, it's good to test different parameters to find a proper solution. The ALMA data suggests you have strong cold dust emission, typical in SF galaxies. What's the X-ray luminosity of this source? If it's >10^42 erg/s, then it's probably an AGN.

For ALMA, yes, what you do is correct. Since there's no real "filters" in ALMA, we often do things like that as an approximation. 

You can generate theoretical SEDs by using the "savefluxes" mode (to fit real data, it is "pdfanalysis" mode) in the pcigale.ini file. Let me know if you have trouble with that.

**Reply by TianmuGao** [reply]
Yeah, I think it's probably a starburst galaxy. I will try the "savefluxes" mode tomorrow to see of I can it.

By the way, the values of some variations in the 'best_model.fits' also seem reasonable. For example, the SFR is about 0.5 (I guess it should be solar mass per year?). They were either too large, or too small before.

Thank you again!

**Reply by TianmuGao** [reply]
Hi! I have a few questions about "savefluxes" mode.

1. What bands and properties should I enter? For bands, I may enter the name of filters I'm using. So what about properties?
2. I'm using the default parameter configuration as an attempt, and there is only one value for each parameter. So why there are 9 models to be calculated?

[pcigale.ini.txt](https://github.com/mboquien/cigale/files/7548274/pcigale.ini.txt)

3. The unit of the flux is still 'mJy', but I think the order of magnitude is wrong(?)

![SED_sim](https://user-images.githubusercontent.com/58971624/142031121-c1b5aa37-0a3c-43b7-b4f0-a25f2f9ace13.jpg)

Thank you!

**Reply by Guang91** [reply]
1. “properties” mean the rest-frame properties you want in the output (e.g., equivalent width). You can leave it empty for your cases, as your interest is to know how parameters affect SEDs.  
2. The 9 models correspond to different alpha_ox values (see Yang+2020). For each parameter set, X-CIGALE always generates 9 models with different alpha_ox if you include the x-ray module. 
3. It's not wrong, as the model is not normalized to real sources. Depending on your science interest, you can normalize it by, e.g., stellar mass or SFR (see the output catalog file models-block-0.fits or models-block-0.txt) or just use the shape while neglecting normalization.

**Reply by TianmuGao** [reply]
Hi! I have two new questions:

1. I have been trying to plot the contributions of each component, but I can't use the 'pcigale-plots' command ('AttributeError: 'Line2D' object has no property 'nonpositive''). Therefore, I'm trying to use the data stored in 'best_model.fits'. Unattenuated stellar continuum is plotted using 'stellar.young' and 'stellar.old'. Unattenuated nebular continuum is also plotted in a similar way. I would like to ask if this is the right way to plot these curves. Also, how can I plot the attenuated continuum?

2. How should I obtain the errors of the physical properties, such as SFR and dust luminosity?

<img width="330" alt="屏幕快照 2021-12-29 下午11 52 34" src="https://user-images.githubusercontent.com/58971624/147680139-0dd4b691-2a8e-4041-ae51-494a6eff6cbf.png">

![best_model](https://user-images.githubusercontent.com/58971624/147679731-7f0e8eb3-2b94-4b2f-ab33-163784751f91.jpg)

**Reply by Guang91** [reply]
>> I have been trying to plot the contributions of each component, but I can't use the 'pcigale-plots' command ('AttributeError: 'Line2D' object has no property 'nonpositive''). 

This might be a bug @mboquien 

>> Therefore, I'm trying to use the data stored in 'best_model.fits'. Unattenuated stellar continuum is plotted using 'stellar.young' and 'stellar.old'. Unattenuated nebular continuum is also plotted in a similar way. I would like to ask if this is the right way to plot these curves. Also, how can I plot the attenuated continuum?

Yes, you can plot this way. There are columns like "attenuation.stellar.old", which are negative values. You can plot attenuated as 'stellar.old' + "attenuation.stellar.old", similar to young and nebula. 

>> How should I obtain the errors of the physical properties, such as SFR and dust luminosity?

The best-fitted values do not have an associated error. 
But the Bayesian values do. To get the Bayesian value and error of certain properties, you need to fill in those property names in pcigale.ini, e.g., 
[analysis_params] 
variables = sfh.sfr, dust.luminosity

**Reply by mboquien** [reply]
> 1. I have been trying to plot the contributions of each component, but I can't use the 'pcigale-plots' command ('AttributeError: 'Line2D' object has no property 'nonpositive'').

@TmGao This is probably because you are using an old version of matplotlib. Updating to matplotlib 3.3 or more recent should fix the issue..

**Reply by TianmuGao** [reply]
I've upgraded matplotlib, and now 'pcigale-plots' works! Thank you all! Well, is there anyway I can customize the axis ranges?

I added those properties to 'variables' and it generated a series of 'chi2-block-0.npy' files. I also find that xcigale provides a tool named 'read_chi.py'. Should I use this to read these files? Besides, I should also set 'save_chi2 = properties', right?

**Reply by Guang91** [reply]
Yes, the range can be customized, see instructions by "pcigale-plots sed --help"

The 'chi2-block-0.npy' contains all the raw chi2 data for each model (can be read by read_chi.py). If you only need the error, you don't need it, b/c the error is already calculated in results.fits. You can set "save_chi2 = none" so that the code will not output 'chi2-block-0.npy' (and this can save you lots of disk space).

**Reply by TianmuGao** [reply]
Thank you! I will check results.fits. Yes, those files really take up lots of space.
